code/game/machinery/slide_projector.dm

```diff
@@ -0,0 +1,189 @@
+/obj/item/storage/slide_projector
+	name = "slide projector"
+	desc = "A handy device capable of showing an enlarged projection of whatever you can fit inside."
+	icon = 'icons/obj/items/device/projector.dmi'
+	icon_state = "projector0"
+	max_w_class = ITEM_SIZE_SMALL
+	max_storage_space = BASE_STORAGE_CAPACITY(ITEM_SIZE_SMALL)
+	use_sound = 'sound/effects/storage/toolbox.ogg'
+	var/list/global/projection_types = list(
+		/obj/item/photo = /obj/effect/projection/photo,
+		/obj/item/paper = /obj/effect/projection/paper,
+		/obj/item = /obj/effect/projection
+	)
+	var/obj/item/current_slide
+	var/obj/effect/projection/projection
+
+/obj/item/storage/slide_projector/Destroy()
+	current_slide = null
+	stop_projecting()
+	. = ..()
+
+/obj/item/storage/slide_projector/on_update_icon()
+	icon_state = "projector[!!projection]"
+
+/obj/item/storage/slide_projector/get_mechanics_info()
+	. = ..()
+	. += "Click in hand to open the interface."
+
+/obj/item/storage/slide_projector/remove_from_storage(obj/item/W, atom/new_location, var/NoUpdate = 0)
+	. = ..()
+	if(. && W == current_slide)
+		set_slide(length(contents) ? contents[1] : null)
+
+/obj/item/storage/slide_projector/handle_item_insertion(var/obj/item/W, var/prevent_warning = 0, var/NoUpdate = 0)
+	. = ..()
+	if(. && !current_slide)
+		set_slide(W)
+
+/obj/item/storage/slide_projector/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if(!current_slide)
+		return
+	project_at(get_turf(target))
+
+/obj/item/storage/slide_projector/proc/set_slide(obj/item/new_slide)
+	current_slide = new_slide
+	playsound(loc, 'sound/machines/slide_change.ogg', 50)
+	if(projection)
+		project_at(get_turf(projection))
+
+/obj/item/storage/slide_projector/proc/check_projections()
+	if(!projection)
+		return
+	if(!(projection in view(7,get_turf(src))))
+		stop_projecting()
+
+/obj/item/storage/slide_projector/proc/stop_projecting()
+	if(projection)
+		QDEL_NULL(projection)
+	GLOB.moved_event.unregister(src, src, .proc/check_projections)
+	update_icon()
+	
+/obj/item/storage/slide_projector/proc/project_at(turf/target)
+	stop_projecting()
+	if(!current_slide)
+		return
+	var/projection_type
+	for(var/T in projection_types)
+		if(istype(current_slide, T))
+			projection_type = projection_types[T]
+			break
+	projection = new projection_type(target)
+	projection.set_source(current_slide)
+	GLOB.moved_event.register(src, src, .proc/check_projections)
+	update_icon()
+
+/obj/item/storage/slide_projector/attack_self(mob/user)
```

Comment:
Minor thought, if you ever made one of those projectors Deer linked in the chat, you could check anchored on attack_hand() to redirect to interact() as well.

Reply:
I just figured that this is already useless enough feature and making it tied ot specific location makes chance no one would ever use it to 100% cue handheld